src/sections/layout/header/LoggedInHeaderActions.tsx

```diff
@@ -35,8 +37,16 @@ export const LoggedInHeaderActions = ({
     })
   }
 
-  const createCollectionRoute = RouteWithParams.CREATE_COLLECTION()
-  const createDatasetRoute = RouteWithParams.CREATE_DATASET()
+  if (!isLoading && !collection) {
+    return <PageNotFound />
```

Comment:
Yes, you're right! I was thinking of <PageNotFound> as a way to redirect to an error page, but of course this is not the place to do it.
I was struggling with how to handle collection being undefined, since as you say that should never happen.  In the js-dataverse use case, it either returns a collection or throws an error.  It would be nice to assert in the component that the collection will never be undefined after it is loaded, but I couldn't figure out a way to do that. Since I can't assert that, I have to handle the undefined case, or typescript will complain.